packages/java-edition/src/mcfunction/tree/argument.ts

```diff
@@ -150,16 +150,46 @@ export interface MinecraftMobEffectArgumentTreeNode
 {
 	parser: 'minecraft:mob_effect'
 }
+export interface NbtParserProperties extends Record<string, unknown> {
+	/**
+	 * The NBT checker should check this argument by dispatching on `dispatcher`
+	 * with a static index inferred from the argument that is at the index
+	 * `dispatchedBy` relative to the current argument and optionally indexing
+	 * on the dispatched type by indices inferred from the argument that is at the
+	 * index `indexedBy` relative to the current argument.
+	 */
+	dispatcher:
+		| 'minecraft:block_entity'
+		| 'minecraft:entity'
+		| 'minecraft:storage'
+	/**
+	 * Could point to a vec3, an entity, an entity resource location, or a storage
+	 * resource location.
+	 *
+	 * @see {@link NbtParserProperties.dispatcher}
+	 */
+	dispatchedBy: number
+	/**
+	 * Could point to an NBT path.
+	 *
+	 * @see {@link NbtParserProperties.dispatcher}
+	 */
+	indexedBy?: number
+	isPredicate?: boolean
```

Comment:
This field could use some documentation. It also doesn't seem to be used.

Reply:
Oh nice I understand what this means now. This is also where `#[canonical]` comes in probably?

Reply:
Probably